expression/aggregation/descriptor.go

```diff
@@ -79,6 +79,27 @@ func (a *AggFuncDesc) Clone() *AggFuncDesc {
 	return &clone
 }
 
+// Split splits `a` into two aggregate descriptors for partial phase and
+// final phase individually.
+// This function is only used when executing aggregate function parallelly.
+func (a *AggFuncDesc) Split() (finalAggDesc *AggFuncDesc) {
+	if a.Mode == CompleteMode {
+		a.Mode = Partial1Mode
+	} else {
+		a.Mode = Partial2Mode
+	}
+	finalAggDesc = &AggFuncDesc{
+		Name:        a.Name,
+		Mode:        FinalMode, // We only support FinalMode now in final phase.
+		HasDistinct: a.HasDistinct,
+		RetTp:       a.RetTp,
+	}
+	if finalAggDesc.Name == ast.AggFuncGroupConcat {
```

Comment:
we also need to modify the arguments for `avg` when the `a` is in `Complete` mode.

Reply:
For avg, we do not need to modify the args here,
since we use `PartialResult` as the intermediate result, 
the args will not be used when evaluating in `Final Phase`.

`sep` will be used for Group_concat

Reply:
Uh, sorry for the wrong comment. I mean we need to modify the output ordinal of `avg`, but `AggFuncDesc` doesn't maintain this info, so it's no need to do it.

Reply:
1. If `a.Mode` is `Complete`, and `a.FuncName` is `avg`, we need to modify the `arg` of `finaAggDesc`
2. we need to set `arg` for other aggregate functions.

Reply:
We do not need to set args for the aggregate function in `FinalMode`,
since args would **not** be used during the final phase,
we directly use PartialResult to evaluate.

Reply:
As I already said, this function can also be used when we building plan, although we are using it for now, we need to make it behaves right in every aspect.